configure-ironic.sh

```diff
@@ -5,11 +5,16 @@ export IRONIC_KEY_FILE=/certs/ironic/tls.key
 export IRONIC_CACERT_FILE=/certs/ca/ironic/tls.crt
 export IRONIC_INSPECTOR_CERT_FILE=/certs/ironic-inspector/tls.crt
 export IRONIC_INSPECTOR_CACERT_FILE=/certs/ca/ironic-inspector/tls.crt
+export MARIADB_CERT_FILE=/certs/mariadb/tls.crt
+export MARIADB_KEY_FILE=/certs/mariadb/tls.key
```

Comment:
from ironic point of view (this file is only for ironic, not mariadb) the only thing we need is either the CA or the actual certificate itself. in order to support seamlessly rotation of certificates from cert-manager, we should rather use the CA. so you can replace the use of both file by the single use of the CA certificate only. And you should refer to that CA in ironic.conf.j2, and that would be a separate CA from Ironic's, like we have a different CA for inspector. If the user wants to use a single CA for all components, they can just mount the same secret multiple times, but we should probably support different CAs for each component.

Reply:
As mTLS is not supported now, even we have a CA for MariaDB, but currently MariaDB only needs to know its key and certificate to enable TLS. So it should be better if Ironic refers to these files to determine whether should it enable TLS with MariaDB. This reference is only to the file location, to check if files exist, but it doesn't care about the content, so I bet it should be fine when the certificate is replaced.